test/error_tests.cpp

```diff
@@ -22,9 +22,9 @@ TEST(ErrorTests, BasicSSLFailure) {
     auto curl_version = curl_version_info(CURLVERSION_NOW);
     auto expected = ErrorCode::UNSUPPORTED_PROTOCOL;
     if(curl_version->features & CURL_VERSION_SSL) {
-        expected = ErrorCode::SSL_CONNECT_ERROR;
+        expected = ErrorCode::CONNECTION_FAILURE;
     }
-    EXPECT_EQ(expected, response.error.code);
+    EXPECT_EQ(expected, response.error.code) << response.error.message;
```

Comment:
Might be me, but what happens in this line if you `<<` a string into the `EXPECT_EQ`?

Reply:
I remember it will be used as [a custom failure message](https://github.com/google/googletest/blob/master/googletest/docs/primer.md#assertions).

> To provide a custom failure message, simply stream it into the macro using the << operator or a sequence of such operators. An example:
```cpp
ASSERT_EQ(x.size(), y.size()) << "Vectors x and y are of unequal length";

for (int i = 0; i < x.size(); ++i) {
  EXPECT_EQ(x[i], y[i]) << "Vectors x and y differ at index " << i;
}
```